### PR TITLE
Client and server gss based negotiated authentication. Authentication moved to http layer and reomved from wsman layer. 

### PR DESCRIPTION
**Pull request by yakman2020**

### CODE REVIEW

File: Unix/http/http.c [hunk 1]
```diff
@@ -660,6 +577,35 @@ static Http_CallbackResult _ReadHeader(
     memcpy( handler->recvPage + 1, data, handler->receivedSize );
     handler->recvingState = RECV_STATE_CONTENT;
 
+/*
```
Comment (lilyfang) [inline Unix/http/http.c]: Any reason that this logic is commented out?

File: Unix/base/user.c [hunk 2]
```diff
@@ -405,8 +405,8 @@ int LookupUser(const char* user, uid_t* uid, gid_t* gid)
     if (s_ignoreAuthCalls)
         return 0;
 
-    if (0 != getpwnam_r(user, &pwd, buf, sizeof(buf), &ppwd) ||
-        !ppwd)
+    int r = getpwnam_r(user, &pwd, buf, sizeof(buf), &ppwd);
```
Comment (niroyb) [inline Unix/base/user.c]: Set errno to 0 before call to getpwnam_r

File: Unix/etc/omiserver.conf [hunk 3]
```diff
@@ -18,12 +18,12 @@
 ##
 ## trace -- enable tracing to standard output (default is 'false')
 ##
-#trace=(true|false)
+trace=true
```
Comment (niroyb) [inline Unix/etc/omiserver.conf]: Undo these changes to the default conf

File: Unix/http/GNUmakefile [hunk 4]
```diff
@@ -3,10 +3,15 @@ include $(TOP)/config.mak
 
 LIBRARY = http
 
+#HTTPTOKEN=httptoken.c
```
Comment (niroyb) [inline Unix/http/GNUmakefile]: Remove

File: Unix/http/GNUmakefile [hunk 5]
```diff
@@ -3,10 +3,15 @@ include $(TOP)/config.mak
...
+
 SOURCES = \
     httpcommon.c \
     http.c \
-    httpclient.c
+    httpauth.c \
+    httpclient.c \
+    httpclientauth.c \
+    $(HTTPTOKEN)
```
Comment (niroyb) [inline Unix/http/GNUmakefile]: Remove

File: Unix/http/http.c [hunk 6]
```diff
@@ -57,6 +57,7 @@ typedef void SSL_CTX;
 #define SSL_accept(c) 0
 
 #endif
+#include "http_private.h"
```
Comment (niroyb) [inline Unix/http/http.c]: Move higher with other includes

File: Unix/http/http.c [hunk 7]
```diff
@@ -368,6 +278,11 @@ static MI_Result _Sock_ReadAux(
             handler->acceptDone = MI_TRUE;
             return _Sock_ReadAux(handler,buf,buf_size,sizeRead);
         }
+	else {
```
Comment (niroyb) [inline Unix/http/http.c]: Fix bracket

File: Unix/http/http.c [hunk 8]
```diff
@@ -697,6 +643,49 @@ static Http_CallbackResult _ReadData(
...
+    Check the authentication/authorization type. It has to be "Basic" or "Negotiate" (That is what OMI supports).
+    In case it is neither, we need to inform the user that this is not supported auth.
+
+    Note: the old behavior was, in the http layer. We check if the auth is not "Basic", then we don't set
+    the username and password, so it will fail here in WSMAN layer, but the error will be 500 error code
+    which means internal server error which doesn't clarify anything to the user. Now we are returning 401
+    which will be interpereted by the client and give a meaningful message.
+    Also, in the wsman specification, it was mentioned that we should return 401 (HTTP_ERROR_CODE_UNAUTHORIZED)
+    with the list of all supported authentication, and they mentioned that this authentication check is prefered
+    to be in the HTTP layer not here but this will be a future change.
```
Comment (niroyb) [inline Unix/http/http.c]: This refactor put the code in the HTTP layer so the comment has to be updated.

**Comment by paulcallen** [comment]
Not sure why it thinks you need to merge all the existing branch items in as well as your own changes. Some rebasing/merging is going to be needed. Hopefully just rebasing though.

**Comment by yakman2020** [comment]
Something got weird with history.
I will generate a pull request from a new branch v-brucc-client-auth-review
